src/Finch_SolidificationData.hpp

```diff
@@ -83,23 +96,15 @@ class SolidificationData
         auto tm =
             Cabana::Grid::createArray<double, memory_space>( "tm", layout );
         tm_view = tm->view();
-
-        folder_name = db.sampling.directory_name;
     }
 
-    void updateEvents()
+    void updateEvents( Grid<memory_space>& grid )
```

Comment:
Why this change instead of storing `local_mesh` locally in the solidification class?

Reply:
I was trying to separate these classes and I figured since we have to extract so many other things may as well not store just some of it (like the the `local_mesh`